core/chains/evm/txmgr/models.go

```diff
@@ -285,20 +304,20 @@ func (e EthTx) GetLogger(lgr logger.Logger) logger.Logger {
 
 // GetChecker returns an EthTx's transmit checker spec in struct form, unmarshalling it from JSON
 // first.
-func (e EthTx) GetChecker() (TransmitCheckerSpec, error) {
+func (e EthTx[ADDR, TX_HASH]) GetChecker() (TransmitCheckerSpec, error) {
 	if e.TransmitChecker == nil {
 		return TransmitCheckerSpec{}, nil
 	}
 	var t TransmitCheckerSpec
 	return t, errors.Wrap(json.Unmarshal(*e.TransmitChecker, &t), "unmarshalling transmit checker")
 }
 
-var _ txmgrtypes.PriorAttempt[gas.EvmFee, common.Hash] = EthTxAttempt{}
+var _ txmgrtypes.PriorAttempt[gas.EvmFee, *evmtypes.TxHash] = EthTxAttempt[*evmtypes.Address, *evmtypes.TxHash]{}
 
-type EthTxAttempt struct {
+type EthTxAttempt[ADDR commontypes.Hashable, TX_HASH commontypes.Hashable] struct {
```

Comment:
Sorry if I missed the thread, but why are we avoiding standard single letters in cases like this?

Reply:
Regardless, is the `TX_` prefix redundant?
```suggestion
type EthTxAttempt[ADDR commontypes.Hashable, HASH commontypes.Hashable] struct {
```

Reply:
If you look at various interfaces in the common/txmgr/types package, the use of generics is increasing.
Single letters are just too confusing. Think about various types:
TxHash, BlockHash, Address, Receipt, Fee, Receipt, ChainId, TxMeta, etc.

I just feel using descriptive types is far more readable, as the same type can be used in various interface, and mean the same thing.

However, if we still like to look at the naming convention for these, I would say, lets wait for everything to be completed and generalized, see how many generic types are remaining at the end, and then get a naming convention that sounds acceptable.

Reply:
I don't necessarily object to the descriptive names  (trying to understand the rule being used first), but redundant descriptions don't add value. In this case, we are already scoped to a tx, so the tx prefix doesn't add information, or distinguish from another hash type.

Reply:
We do have BLOCK_HASH and TX_HASH in Txmgr, Broadcaster and Confirmer, StorageService.
So i preferred to use these names everywhere, for consistency.

Reply:
This is not very go like. Using broad namespaces in narrow scopes is needlessly verbose, and creates confusion for readers who see repeated/redundant/unnecessary modifiers.

https://go.dev/doc/effective_go#package-names

> The importer of a package will use the name to refer to its contents, so exported names in the package can use that fact to avoid repetition. (Don't use the import . notation, which can simplify tests that must run outside the package they are testing, but should otherwise be avoided.) For instance, the buffered reader type in the bufio package is called Reader, not BufReader, because users see it as bufio.Reader, which is a clear, concise name. Moreover, because imported entities are always addressed with their package name, bufio.Reader does not conflict with io.Reader. Similarly, the function to make new instances of ring.Ring—which is the definition of a constructor in Go—would normally be called NewRing, but since Ring is the only type exported by the package, and since the package is called ring, it's called just New, which clients of the package see as ring.New. Use the package structure to help you choose good names.